src/Compilers/Core/Portable/Emit/EditAndContinue/SymbolChanges.cs

```diff
@@ -51,21 +50,67 @@ protected SymbolChanges(DefinitionMap definitionMap, IEnumerable<SemanticEdit> e
+        public ImmutableDictionary<ISymbolInternal, ImmutableArray<ISymbolInternal>> GetAllDeletedMembers()
         {
             var builder = ImmutableDictionary.CreateBuilder<ISymbolInternal, ImmutableArray<ISymbolInternal>>();
 
-            foreach (var type in _deletedMembers)
+            foreach (var (type, deletedMembers) in _deletedMembers)
             {
-                if (GetISymbolInternalOrNull(type.Key) is { } typeSymbol)
+                if (GetISymbolInternalOrNull(type) is not { } typeSymbol)
                 {
-                    builder.Add(typeSymbol, ToInternalSymbolArray(type.Value));
+                    continue;
                 }
+
+                var internalSymbols = GetDeletedMemberInternalSymbols(deletedMembers, includeMethods: true, includeProperties: true, includeEvents: true);
+
+                builder.Add(typeSymbol, internalSymbols);
+
             }
 
             return builder.ToImmutable();
         }
 
+        private ImmutableArray<ISymbolInternal> GetDeletedMemberInternalSymbols(ISet<ISymbol> deletedMembers, bool includeMethods, bool includeProperties, bool includeEvents)
+        {
+            var internalSymbols = ArrayBuilder<ISymbolInternal>.GetInstance();
+
+            foreach (var symbol in deletedMembers)
+            {
+                if (GetISymbolInternalOrNull(symbol) is { } internalSymbol)
+                {
+                    if (includeProperties &&
+                        symbol is IMethodSymbol { AssociatedSymbol: IPropertySymbol propertySymbol } &&
+                        (propertySymbol.GetMethod is null || propertySymbol.GetMethod == symbol))
+                    {
+                        var internalPropertySymbol = GetISymbolInternalOrNull(propertySymbol);
+                        if (internalPropertySymbol is not null)
+                        {
+                            internalSymbols.Add(internalPropertySymbol);
+                        }
+                    }
+
+                    if (includeEvents &&
+                        symbol is IMethodSymbol { AssociatedSymbol: IEventSymbol eventSymbol } &&
+                        eventSymbol.AddMethod == symbol)
+                    {
+                        var internalEventSymbol = GetISymbolInternalOrNull(eventSymbol);
+                        if (internalEventSymbol is not null)
+                        {
+                            internalSymbols.Add(internalEventSymbol);
+                        }
+                    }
+
+                    if (includeMethods &&
+                        symbol is IMethodSymbol)
+                    {
+                        internalSymbols.Add(internalSymbol);
+                    }
+                }
+            }
+
+            return internalSymbols.ToImmutableAndFree();
+        }
+
         public ImmutableArray<ISymbolInternal> GetDeletedMethods(IDefinition containingType)
```

Comment:
>GetDeletedMethods [](http://example.com/codeflow?start=47&length=17)

It looks like `GetDeletedMethods(IDefinition)`, `GetDeletedProperties(IDefinition)`, and `GetDeletedEvents(IDefinition)` only differ in the arguments passed to `GetDeletedMemberInternalSymbols()`. Consider extracting a common `GetDeletedMembers(IDefinition containingType, bool includeMethods, bool includeProperties, bool includeEvents)` helper that all three could use.